core/wiki/macros/keyboard-driven-input.tid

```diff
@@ -0,0 +1,53 @@
+title: $:/core/macros/keyboard-driven-input
+tags: $:/tags/Macro
+
+\define keyboard-input-actions()
+<$list filter="[<__index__>match[]]">
+<$action-setfield $tiddler=<<__storeTitle__>> text={{{ [<__tiddler__>get<__field__>] }}}/>
+</$list>
+<$list filter="[<__index__>!match[]]">
+<$action-setfield $tiddler=<<__storeTitle__>> text={{{ [<__tiddler__>getindex<__index__>] }}}/>
+</$list>
+\end
+
+\define keyboard-input-up-down-actions(reverse,afterOrBefore) <$macrocall $name="input-next-actions" filter1=<<__filter1__>> filter2=<<__filter2__>> refreshTitle=<<__refreshTitle__>> afterOrBefore="$afterOrBefore$" reverse="$reverse$" tiddler=<<__tiddler__>> storeTitle=<<__storeTitle__>> field=<<__field__>> index=<<__index__>> filterMinLength=<<__filterMinLength__>> selectionStateTitle=<<__selectionStateTitle__>>/>
+
+\define input-next-actions(afterOrBefore:"after",reverse:"",tiddler,storeTitle,field,index,filter1,filter2,refreshTitle,selectionStateTitle)
+<$list filter="[<__storeTitle__>get[text]minlength<__filterMinLength__>] [<__filterMinLength__>match[0]] +[limit[1]]" variable="ignore">
+<$vars userInput={{{ [<__storeTitle__>get[text]] }}} selectedItem={{{ [<__selectionStateTitle__>get[text]] }}}>
+<$set name="filteredList" filter="[subfilter<__filter1__>addsuffix[-list1]] =[subfilter<__filter2__>addsuffix[-list2]]">
+<$set name="nextItem" value={{{ [enlist<filteredList>$afterOrBefore$<selectedItem>] ~[enlist<filteredList>$reverse$nth[1]] }}}>
+<$list filter="[<nextItem>minlength[1]]">
+<$action-setfield $tiddler=<<__selectionStateTitle__>> text=<<nextItem>>/>
+<$list filter="[<__index__>match[]]">
+<$action-setfield $tiddler=<<__tiddler__>> $field=<<__field__>> $value={{{ [<nextItem>] +[splitregexp[(?:.(?!-))+$]] }}}/>
+</$list>
+<$list filter="[<__index__>!match[]]">
+<$action-setfield $tiddler=<<__tiddler__>> $index=<<__index__>> $value={{{ [<nextItem>] +[splitregexp[(?:.(?!-))+$]] }}}/>
+</$list>
+<$action-setfield $tiddler=<<__refreshTitle__>> text="yes"/>
+</$list>
+</$set>
+</$set>
+</$vars>
+</$list>
+\end
+
+\define keyboard-driven-input(tiddler,storeTitle,field:"text",index:"",tag:"input",type,focus:"",inputAcceptActions,inputAcceptVariantActions,inputCancelActions,placeholder:"",default:"",class,filter1,filter2,focusPopup,rows,minHeight,tabindex,size,autoHeight,filterMinLength:"0",refreshTitle,selectionStateTitle)
+<$keyboard key="((input-accept))" actions=<<__inputAcceptActions__>>>
+<$keyboard key="((input-accept-variant))" actions=<<__inputAcceptVariantActions__>>>
+<$keyboard key="((input-up))" actions=<<keyboard-input-up-down-actions "reverse[]" "before">>>
+<$keyboard key="((input-down))" actions=<<keyboard-input-up-down-actions "" "after">>>
```

Comment:
can we call `input-next-actions` directly here?

Reply:
yes, changed it! thanks